src/transformers/integrations.py

```diff
@@ -163,11 +163,21 @@ def _objective(trial, local_trainer, checkpoint_dir=None):
             local_trainer._tune_save_checkpoint()
             ray.tune.report(objective=local_trainer.objective, **metrics, done=True)
 
+    if not trainer._memory_tracker.skip_memory_metrics:
+        from .trainer_utils import TrainerMemoryTracker
+
+        logger.warning(
+            "Memory tracking for your Trainer is currently "
+            "enabled. Automatically disabling the memory tracker "
+            "since the memory tracker is not serializable."
+        )
+        trainer._memory_tracker = TrainerMemoryTracker(skip_memory_metrics=True)
```

Comment:
@LysandreJik maybe we can have this as a more generic API? nice to have, non blocking